### PR TITLE
2.0.2 return non zero for bad usage

### PR DESCRIPTION
**Pull request by corwinsheahan-wf**
## Motivation
  
Back patch of [this PR](https://github.com/Workiva/dependency_validator/pull/78) to the 2.x release line.

## Changes
  
* Cherry pick changes from that ^ PR.
* Ran format under dart 2.14.4
## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Client Platform member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/dependency_validator/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/dependency_validator/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Review by evanweible-wf — COMMENTED** [review]
@corwinsheahan-wf could you add a changelog entry?

**Comment by corwinsheahan-wf** [comment]
@evanweible-wf Done

**Review by evanweible-wf — APPROVED** [review]

File: bin/dependency_validator.dart [hunk 1]
```diff
@@ -16,11 +16,13 @@ import 'dart:io' show exit, stderr, stdout;
 
 import 'package:args/args.dart';
 import 'package:dependency_validator/dependency_validator.dart';
+import 'package:io/io.dart';
```
Comment (evanweible-wf) [inline bin/dependency_validator.dart:19]: This isn't a dependency in v2, so we need to add it:

```yaml
  io: ^0.3.5
```

**Review by evanweible-wf — CHANGES_REQUESTED** [review]

File: bin/dependency_validator.dart (hunk 1 above)
Comment (corwinsheahan-wf) [reply bin/dependency_validator.dart:19]: done

**Review by evanweible-wf — APPROVED** [review]

**Comment by evanweible-wf** [comment]
QA +1
- [x] Ran tests locally and they passed (after adding the `meta: <1.7.0` workaround which is only needed for local dev)

@Workiva/release-management-p

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**